### PR TITLE
Add fn:xml-to-json()

### PR DESCRIPTION
**Pull request by agh2342**
Add documentation for fn:xml-to-json().

### CODE REVIEW

**Comment by agh2342** [comment]
I don't get this to render correctly in the final xar.
It seems like the markdown parser does not escape the `<string>` tags inside the code blocks.
It seems like someone is replacing backslash characters in the backticked areas.
Any suggestions?

**Comment by duncdrum** [comment]
@agh2342 this sounds like a bug in the markdown package, you might have open an issue there.

**Comment by agh2342** [comment]
Looks like there is an open bug report already. (https://github.com/eXist-db/exist-markdown/issues/4)

File: src/main/xar-resources/data/docs/fn/xml-to-json.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# fn:xml-to-json() ([specification](https://www.w3.org/TR/xpath-functions-31/#func-xml-to-json"))
+handles most valid input according to specification. Notable exceptions follow.
+
+All (un-)escaping is delegated to com.fasterxml.jackson. All output is generated by jackson
```
Comment (duncdrum) [inline src/main/xar-resources/data/docs/fn/xml-to-json.md]:
```suggestion
All output is generated by the `jackson` library and should be proper JSON. All (un-) escaping is delegated to `com.fasterxml.jackson`. 
```

File: src/main/xar-resources/data/docs/fn/xml-to-json.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+and should be proper JSON.
+
+`fn:xml-to-json()` does not replace codepoints in the range `1-31` or `127-159` in any string or map key.
+
+`fn:xml-to-json()` unescapes and reescapes any string and map key marked as being escaped.
+It does not do additional special character replacements mentioned in the spec.
```
Comment (duncdrum) [inline src/main/xar-resources/data/docs/fn/xml-to-json.md]: please put a link to the spec on `spec`

File: src/main/xar-resources/data/docs/fn/xml-to-json.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+It does not do an otherwise verbatim copy.
+
+# Examples
```
Comment (duncdrum) [inline src/main/xar-resources/data/docs/fn/xml-to-json.md]: Yes this is much better. The examples actually explain the known limitations in way that is easy to understand. This is why code listings are often better then prose explanations.

File: src/main/xar-resources/data/docs/fn/xml-to-json.md (hunk 1 above)
Comment (joewiz) [reply src/main/xar-resources/data/docs/fn/xml-to-json.md]: @duncdrum's suggestion looks good to me here. @agh2342 Could you please accept or reject the suggestion when you have the chance?

File: src/main/xar-resources/data/docs/fn/xml-to-json.md (hunk 2 above)
Comment (joewiz) [reply src/main/xar-resources/data/docs/fn/xml-to-json.md]: The spec is already linked to a few lines above in the head of the document, with the label "specification."

**Comment by joewiz** [comment]
@duncdrum Let's merge this PR and handle the markdown parser issue separately. The PR is fine.

**Comment by joewiz** [comment]
p.s.  I plan to raise the topic of the markdown parser in the next community call—to discuss development of a new wrapper module around one of the java implementations of commonmark parser.

**Comment by duncdrum** [comment]
This whole funcdoc detour is on @line-o so it’s his PR to merge.

**Comment by joewiz** [comment]
@line-o @agh2342 Please just accept/reject the suggestion above, then merge at will!

**Pull request merged**